File: services/registration-service/src/main/java/com/ufma/tap/registration/model/Project.java
// Caminho: services/registration-service/src/main/java/com/ufma/tap/registration/model/Project.java
package com.ufma.tap.registration.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import jakarta.persistence.FetchType;
import java.util.List;

@Entity
@Table(name = "projects")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Project {
    @Id
    private String id;
    private String name;
    private String location;
    private String region;
    @ElementCollection(fetch = FetchType.EAGER) // <<< ADICIONE ISSO AQUI
    @CollectionTable(name = "project_supported_brokers", joinColumns = @JoinColumn(name = "project_id"))
    @Column(name = "broker_name")
    private List<String> supportedBrokers;

    @Transient
    private String authToken;
}

File: front-end-flutter/lib/models/project.dart
class Project {
  final String id;
  final String name;
  final String location;
  final String region;
  final List<String> supportedBrokers;
  final String token;

  Project({
    required this.id,
    required this.name,
    required this.location,
    required this.region,
    required this.supportedBrokers,
    required this.token,
  });

  factory Project.fromJson(Map<String, dynamic> json) {
    return Project(
      id: json['id'] ?? '',
      name: json['name'] ?? '',
      location: json['location'] ?? '',
      region: json['region'] ?? '',
      supportedBrokers: List<String>.from(json['supportedBrokers'] ?? []),
      token: json['authToken'] ?? '', // Backend retorna 'authToken'
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'location': location,
      'region': region,
      'supportedBrokers': supportedBrokers,
      'token': token,
    };
  }
}
